daemon/kill.go

```diff
@@ -139,11 +150,10 @@ func (daemon *Daemon) Kill(container *containerpkg.Container) error {
 	}
 
 	// 1. Send SIGKILL
-	if err := daemon.killPossiblyDeadProcess(container, syscall.SIGKILL); err != nil {
-		// kill failed, check if process is no longer running.
-		if errors.As(err, &errNoSuchProcess{}) {
-			return nil
-		}
+	err := daemon.killPossiblyDeadProcess(container, syscall.SIGKILL)
+	if err == nil {
+		// SIGKILL was sent successfully, or process was already gone. We're done.
+		return nil
```

Comment:
Hm... right, so I _think_ I understand the problem. Previously we were returning early on an _ERROR_  but only if that error was `errNoSuchProcess`. But if there was _NO_ error, we would continue the code.

So I think there's a change in logic in my first 2 commits, which could either be "correct" (but code doesn't handle it well), or incorrect. Need to do some more digging

I'll move the other two commits to a separate PR